### ISSUE TITLE
DeplymentController uses an unusual construct for getByKey

### ISSUE THREAD

**Issue by soltysh** [openshift/origin#15928]
I found this https://github.com/openshift/origin/blob/0c81307bf6eb180d27427e67bb9f0996426e04f8/pkg/deploy/controller/deployer/factory.go#L219-L222 which I had to fix in image controller in https://github.com/openshift/origin/pull/15927.

I'm guessing @tnozicka you'll be the best person to address the same issue in deployments. Generally `getByKey` should not interpret errors.

**Comment by tnozicka** [comment]
Yes, that seems ugly. Not sure if that is a bug since it's a private function so as long as the deployer controller checks for it properly, should work; although it's error prone.

**Comment by soltysh** [comment]
> although it's error prone.

That's why you need to fix it. I did the same in images.
